app/controllers/api/v1/mixins/destroy_mixin.rb

```diff
@@ -3,7 +3,10 @@ module V1
     module Mixins
       module DestroyMixin
         def destroy
-          model.destroy(params.require(:id))
+          record = model.find(params.require(:id))
+          authorize(record)
```

Comment:
I'm guessing `authorize(record)` raises an exception if there is an illegal attempt? 

Otherwise was wondering what stops `record.destroy` from getting executed.

Reply:
Yep - exactly. Pundit raises `Pundit::NotAuthorized` when the authorization fails.